### ISSUE TITLE
Workflow giving 404 when using cosmosdb as Actor state store

### ISSUE THREAD

**Issue by DeepanshuA** [dapr/dapr#6647]
## In what area(s)?

Had a conversation with @cgillum, in which he pointed out that using Cosmos db may be an issue right now with workflows.
Tried using cosmosdb and redis as actor state stores respectively for exactly same scenarios.
1. With redis, 12 keys were persisted and no error was received.

Keys:
```
 1) "actors||dapr.internal.default.orderapp.workflow"
 2) "orderapp||dapr.internal.default.orderapp.workflow||f276f281dd1a48fb859e37484919ff0f||metadata"
 3) "orderapp||dapr.internal.default.orderapp.workflow||f276f281dd1a48fb859e37484919ff0f||history-000004"
 4) "orderapp||dapr.internal.default.orderapp.workflow||f276f281dd1a48fb859e37484919ff0f||history-000005" 
 5) "orderapp||dapr.internal.default.orderapp.workflow||f276f281dd1a48fb859e37484919ff0f||history-000003"
 6) "actors||dapr.internal.default.orderapp.activity||metadata"
 7) "orderapp||dapr.internal.default.orderapp.workflow||f276f281dd1a48fb859e37484919ff0f||history-000002"
 8) "orderapp||dapr.internal.default.orderapp.workflow||f276f281dd1a48fb859e37484919ff0f||history-000000"
 9) "orderapp||dapr.internal.default.orderapp.workflow||f276f281dd1a48fb859e37484919ff0f||history-000001"
10) "orderapp||dapr.internal.default.orderapp.workflow||f276f281dd1a48fb859e37484919ff0f||customStatus"
11) "actors||dapr.internal.default.orderapp.workflow||metadata"
12) "actors||dapr.internal.default.orderapp.activity"
```
2. While with cosmosdb, 10 keys were persisted and some 404 errors were noticed.
Keys:
```
 1) "actors||dapr.internal.default.orderapp.workflow"
 2) "actors||dapr.internal.default.orderapp.workflow||metadata"
 3) "orderapp||dapr.internal.default.orderapp.workflow||01f36664c64d4f48802d985a42dfb202||customStatus"
 4) "orderapp||dapr.internal.default.orderapp.workflow||01f36664c64d4f48802d985a42dfb202||metadata"
 5) "actors||dapr.internal.default.orderapp.activity"
 6) "actors||dapr.internal.default.orderapp.activity||metadata"
 7) "orderapp||dapr.internal.default.orderapp.workflow||01f36664c64d4f48802d985a42dfb202||history-000000"
 8) "orderapp||dapr.internal.default.orderapp.workflow||01f36664c64d4f48802d985a42dfb202||history-000001"
 9) "orderapp||dapr.internal.default.orderapp.workflow||01f36664c64d4f48802d985a42dfb202||history-000002"
10) "orderapp||dapr.internal.default.orderapp.workflow||01f36664c64d4f48802d985a42dfb202||inbox-000000"
```
Error:
```
time="2023-07-11T19:07:18.1628726+05:00" level=error msg="Transaction failed due to operation 1 which failed with status code 404" app_id=orderapp component="statestore-actors (state.azure.cosmosdb/v1)" instance=LAPTOP-J62MA7E9 scope=dapr.contrib type=log ver=unknown
time="2023-07-11T19:07:18.1629589+05:00" level=error msg="01f36664c64d4f48802d985a42dfb202: execution failed with a non-recoverable error: transaction failed due to operation 1 which failed with status code 404" app_id=orderapp instance=LAPTOP-J62MA7E9 scope=dapr.runtime.wfengine type=log ver=unknown
```

/area runtime

> /area operator

> /area placement

> /area docs

> /area test-and-release

## What version of Dapr?

1.11

## Expected Behavior

Same keys should have got persisted with cosmosdb also and 404 error shouldn't have received.
## Steps to Reproduce the Problem

Try running workflow python quickstart, with cosmosdb as actorStateStore.

## Release Note

RELEASE NOTE:

**Comment by berndverst** [comment]
What exactly happens in that transaction itself? What is the first operation?

We have plenty of CosmosDB transaction tests otherwise, so this seems to be specific to the transaction being attempted by Workflows.

Also, not sure if this comes up - CosmosDB has a 1MB entity size limit.

**Comment by RyanLettieri** [comment]
This is due to how we are storing/retriving the timer from cosmosDB. In the test, we have a timer that elapses after 3m20s. After this timer expires, we see the 404 error. It is worth noting that this error does not occur when using Redis, nor does it occur when using postgres.

**Comment by ItalyPaleAle** [comment]
Fixed by #6826
